sdk/core/core-http/lib/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -32,21 +41,21 @@ export function bearerTokenAuthenticationPolicy(credential: TokenCredential, sco
  *
  */
 export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
-  private cachedToken: AccessToken | undefined = undefined;
-
   /**
    * Creates a new BearerTokenAuthenticationPolicy object.
    *
    * @param nextPolicy The next RequestPolicy in the request pipeline.
    * @param options Options for this RequestPolicy.
    * @param credential The TokenCredential implementation that can supply the bearer token.
    * @param scopes The scopes for which the bearer token applies.
+   * @param tokenCache The cache for the most recent AccessToken returned from the TokenCredential.
    */
   constructor(
     nextPolicy: RequestPolicy,
     options: RequestPolicyOptions,
     private credential: TokenCredential,
     private scopes: string | string[],
+    private tokenCache: AccessTokenCache
```

Comment:
Out of curiosity, what part will manage the cache?

Reply:
The consuming policy, though right after sending this I realized that there's probably value in sharing token expiration logic in a more legitimate class instead of a simple object. Thoughts?

Reply:
I agree that there's definitely value in centralizing token expiration management, otherwise we will end up with a bug somewhere, or at least different implementations to a small extent. I wonder what would be the benefits of distributed token expiration logic, if that route is decided.